AutoAnswer/AutoAnswer/AppDelegate.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  AppDelegate.swift
+//  AutoAnswer
+//
+//  Created by Minhyuk Kim on 2021/06/11.
+//
+
+import UIKit
+import SendBirdCalls
+import Commons
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+    var window: UIWindow?
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        let appId: String = "94271307-58BC-45F1-9649-F078A2C1A08A"
+        SendBirdCall.configure(appId: appId)
+        SendBirdCall.addDelegate(self, identifier: "AppDelegate")
+        
+        remoteNotificationsRegistration(application)
+        return true
+    }
+    
+    func remoteNotificationsRegistration(_ application: UIApplication) {
+        application.registerForRemoteNotifications()
+        
+        let center = UNUserNotificationCenter.current()
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { success, error in
+            guard error == nil else {
+                print("Error while requesting permission for notifications.")
+                return
+            }
+            
+            // If success is true, the permission is given and notifications will be delivered.
+        }
+    }
+    
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        UserDefaults.standard.remotePushToken = deviceToken
+        SendBirdCall.registerRemotePush(token: deviceToken, completionHandler: nil)
+    }
+    
+    func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        SendBirdCall.application(application, didReceiveRemoteNotification: userInfo)
+    }
+}
+
+// MARK: - Process incoming calls
+extension AppDelegate: SendBirdCallDelegate {
+    func didStartRinging(_ call: DirectCall) {
```

Comment:
I thought that the delegate method runs on main thread, doesn't it?

Reply:
didStartRinging rings on the thread that it was called in. Because it is called by notifications, it can be not main thread.

Reply:
I see. Thanks!